development for project5

AaronMoroney opened the pull request.

AaronMoroney committed:
Initial commit

AaronMoroney committed:
Landing page (all products) see description

All products added dynamically using JS, see // in index.js for description of methodology - css improvements on card height for desktop, mobile and tablet

AaronMoroney committed:
JS logic for product page (individual)

Created individual product page and populated information dynamically - couldn't access image via 'class' on productfront.html so i added an id to it (see productFront.HTML)

commented out logic currently being worked on for options list

AaronMoroney committed:
dropdown selecta! - completed

dropdown selector now installed on the page.

lines 63-77 productFront.js

AaronMoroney committed:
Cart function after mentor session

Debugged for cart function to now receive information from the quantity and colors. follow comments on each line to find the logic of each item.

106 - 117

AaronMoroney committed:
now finding product quantity

A lot of work went into finding the logic of the cart operation - finding the quantity of the selected cart items

AaronMoroney committed:
updated cart functionality

cart

AaronMoroney committed:
populated information

populated information on cart page
-
need to debug - multiples, 'alert?'

AaronMoroney committed:
Appenditure order change

AaronMoroney committed:
Cart Page Quantity Increase

cart page quantity increase now functioning

AaronMoroney committed:
File restructuring *see note*

restructured files to cartRough.js - for rough coding work, and to cart2.js, which is the final cart file - for educational purposes.

AaronMoroney committed:
cart2 - retrieve data

retrieve and manipulate data

AaronMoroney committed:
Cart Post REQ now working

need to access confirmation page

AaronMoroney committed:
changes before debugging session

changes before debugging session with ambassador Nuomodong

AaronMoroney committed:
After debugging session, confirmation page now linked

confirmation page linked

AaronMoroney committed:
Done *check minor bug list below**

Minor bugs - delete and add quantity

AaronMoroney committed:
!! need to fix quantity bug !!

AaronMoroney committed:
Cart quantity update debug

AaronMoroney committed:
(Final Bug) all items quantity

Fixed

AaronMoroney committed:
Folder Restructuring

Project Complete

AaronMoroney committed:
cart name change

above

AaronMoroney committed:
Merge branch 'main' of https://github.com/AaronMoroneyDesignCo/OCR_P5_aaronMoroney into Dev

fpatelm commented on code/front/js/index.js:
```diff
@@ -0,0 +1,93 @@
+** index page (all products)
+*/
+
+const uri = 'http://localhost:3000/api/products/';
```
best practice to declare global const variables all upper case, in this case:
`const URI = 'http://localhost:3000/api/products/';`

fpatelm commented on code/front/js/index.js:
```diff
@@ -0,0 +1,93 @@
+*/
+
+const uri = 'http://localhost:3000/api/products/';
+const singleProductLink = './productFront.html?=';
```
best practice to declare global const variables all upper case, in this case:
const SINGLE_PRIODUCT_LLINK= './productFront.html?=';

fpatelm commented on code/front/js/index.js:
```diff
@@ -0,0 +1,93 @@
+
+function createProductCardsInfo(array) {
+    const length = array.length;
+    for (let i = 0; i<length; i++) {
```
```
 for (let i = 0; i<array.length; i++) {
}

```

fpatelm replied:
you did not update it

AaronMoroney replied:
now updated

fpatelm commented on code/front/js/index.js:
```diff
@@ -0,0 +1,93 @@
+    article.appendChild(productName);
+    article.appendChild(productDescription);
+    
+    return createProductCardView;
```
why are you returning this view that is not used in the calling method?
see line #28, ` createProductCardView(array[i]);`
you are not using the returned value.

fpatelm replied:
@AaronMoroneyDesignCo  this was not resolved!

AaronMoroney replied:
Also had been resolved - is it standard to hit command + S before a push to branch?

fpatelm commented on code/front/js/index.js:
```diff
@@ -0,0 +1,93 @@
+    return createProductCardView;
+}
+
+
```
remove all this empty lines

fpatelm replied:
you did not remove empty lines fo code

AaronMoroney replied:
Also had been previously resolve on my local codebase

fpatelm commented on code/front/js/productFront.js:
```diff
@@ -0,0 +1,148 @@
+let splitForId = queryString.slice(5);
+console.log(splitForId);
+
+const newUri = 'http://localhost:3000/api/products/' + splitForId;
```
make it:
`const PRODUCT_URI  = 'http://localhost:3000/api/products/' + PRODUCT_ID;`

fpatelm commented on code/front/js/productFront.js:
```diff
@@ -0,0 +1,148 @@
+//get _id of product from query parameters
+const queryString = window.location.search;
+console.log(queryString);
+let splitForId = queryString.slice(5);
```
make it
`const PRODUCT_ID = queryString.slice(5);`

fpatelm commented on code/front/js/productFront.js:
```diff
@@ -0,0 +1,148 @@
+*/
+
+//fetch the new URI from get request
+fetch(newUri) 
```
`fetch(PRODUCT_URI)`

AaronMoroney replied:
This was resolved but didn't successfully appear in the git push to dev branch, Perhaps I should hit command+s first on the files

fpatelm commented on code/front/js/productFront.js:
```diff
@@ -0,0 +1,148 @@
+        console.log(data)
+        createIndividualProductView(data);
+        populateDropdown(data);
+        product = data;
```
` product = data; `should be the first thing!

```
console.log(data)
product = data;
createIndividualProductView(data);
populateDropdown(data);
```

fpatelm commented on code/front/js/productFront.js:
```diff
@@ -0,0 +1,148 @@
+    //append
+    imgParent.appendChild(individualProductImg);
+
+    return createIndividualProductView;
```
again,, you are returning a value and not using it ....., see line #23, if you do use it at the calling method, don't return it.

read about java functions, your evaluator will ask this, and you can fail the presentation if you don't know how to explain when to use a return value or not.

fpatelm commented on code/front/js/productFront.js:
```diff
@@ -0,0 +1,148 @@
+});
+
+/*
+** | add to shopping cart
```
add more meaningful comments in all the functions:
what does the function receive?
what does the function do?
what does the function return?

fpatelm commented on code/front/js/productFront.js:
```diff
@@ -0,0 +1,148 @@
+  }
+}
+
+
```
r3emove extra empty lines

AaronMoroney committed:
restructuring - commnet / const etc.

AaronMoroney committed:
Typo index.js

fixed

AaronMoroney committed:
Fix loop

AaronMoroney committed:
closest | delete cart item to 0 bug

When items are deleted from cart - and the cart is empty, the total articles and cost remained visable in the front end - error handling, if else - removed this bug.

Also, instead of parent.parent. I used element.closest to target relevant parents.

fpatelm commented on code/front/js/cart.js:
```diff
@@ -113,7 +113,7 @@ function populateCartInfo(cart) {
     //totals
     //only need 1, class returns an array of hmtl so, [0]
-    let cartPrice = document.getElementsByClassName('cart__price')[0];
+    document.getElementsByClassName('cart__price')[0];
```
this code is not doing anything

fpatelm replied:
you can just remove this line of code

fpatelm commented on code/front/js/cart.js:
```diff
@@ -205,8 +228,13 @@ function updateQuantityCartItem () {
             for (let i = 0; i < cart.length; i++) {
-                itemQuantity.value = event.target.value;
-                cart[i].quantity = itemQuantity[i].value;
+                if (cart.length === 1) {
```
here you have to use the same approach as for deletecart item

imagine you update the "fat milk 2l" item on the cart and on the cart you have as well "fat milk 1l". You need to know which item you want to update the quantity. You need to get the respective item by id and color from the cart and update its quantity.
```
let updatedItem = event.target;
 let color = updatedItem.closest('.cart__item').getAttribute('data-color');
let productId = updatedItem.closest('.cart__item').getAttribute('data-id');
```

AaronMoroney replied:
Hey Faizal, Just to clarify - if I add one product ID with two different colour options -  (a) white (b) black - the product Total in cart = 2. The products are stored in the cart array as two separate objects(products) like so - [  {productID: "1" colour: "white"}, { productID: "2" color: "black" } ] 

Is it necessary to target the data-ID attribute, and the data-color attribute ? I can successfully update the products in the cart without doing so as it stands - and all tests pass.

Will this refactoring suggestion will allow me **not** to have to wrap the logic for updateQuantityCartItem in if else? I don't understand.

fpatelm replied:
will mark this as unnecessary as we discussed in the workplace
